File: core/src/main/java/com/codebasecensus/core/Codebase.java
package com.codebasecensus.core;

import java.net.URI;

/**
 * Created by kevin on 08/12/18 at 11:58
 */
public interface Codebase {
    void cloneRepo(URI uri, CloneCallback callback);

    void cloneRepo(URI uri, String branch, CloneCallback callback);
}

File: core/src/main/java/com/codecensus/core/Codebase.kt
package com.codecensus.core;

import com.codecensus.core.callbacks.AnalyzeCallback
import com.codecensus.core.callbacks.SetupCallback

/**
 * Created by kevin on 08/12/18 at 11:58
 */
interface Codebase {
    fun setup(callback: SetupCallback)

    fun analyzeTotalFileCount(callback: AnalyzeCallback)

    fun contributors()
}